### ISSUE TITLE
puma does not restart after deploy that changes only environment variables

### ISSUE THREAD

**Issue by kzkn** [ajgon/opsworks_ruby#281]
### Expected Behavior
Restart puma workers.

### Actual Behavior
Does not restart puma workers. Deploying again will restart puma workers.
Here is a syslog when puma did not restart.

```
Apr 07 17:14:42 myapp-server sshd[8395]: Received disconnect from 124.223.52.238 port 53812:11: Bye Bye [preauth]
Apr 07 17:14:42 myapp-server sshd[8395]: Disconnected from authenticating user root 124.223.52.238 port 53812 [preauth]
Apr 07 17:14:43 myapp-server monit[8503]: Skipping 'allow localhost' -- host resolved to [::ffff:127.0.0.1] which is present in ACL already
Apr 07 17:14:43 myapp-server monit[8503]: Reinitializing monit daemon
Apr 07 17:14:43 myapp-server monit[931]: Reinitializing Monit -- control file '/etc/monit/monitrc'
Apr 07 17:14:43 myapp-server monit[8509]: Skipping 'allow localhost' -- host resolved to [::ffff:127.0.0.1] which is present in ACL already
Apr 07 17:14:43 myapp-server monit[8509]: Reinitializing monit daemon
Apr 07 17:14:43 myapp-server monit[931]: Skipping 'allow localhost' -- host resolved to [::ffff:127.0.0.1] which is present in ACL already
Apr 07 17:14:43 myapp-server monit[931]: 'myapp-server' Monit reloaded
Apr 07 17:14:56 myapp-server sshd[8587]: Received disconnect from 124.223.52.238 port 54848:11: Bye Bye [preauth]
Apr 07 17:14:56 myapp-server sshd[8587]: Disconnected from authenticating user root 124.223.52.238 port 54848 [preauth]
Apr 07 17:15:01 myapp-server CRON[8643]: pam_unix(cron:session): session opened for user root by (uid=0)
Apr 07 17:15:01 myapp-server CRON[8644]: (root) CMD (flock --nonblock /var/lib/aws/opsworks/lockrun.lock /opt/aws/opsworks/current/bin/opsworks-agent-updater)
Apr 07 17:15:01 myapp-server CRON[8643]: pam_unix(cron:session): session closed for user root
Apr 07 17:15:12 myapp-server sshd[8719]: Received disconnect from 124.223.52.238 port 55860:11: Bye Bye [preauth]
Apr 07 17:15:12 myapp-server sshd[8719]: Disconnected from authenticating user root 124.223.52.238 port 55860 [preauth]
Apr 07 17:15:12 myapp-server crontab[8769]: (deploy) LIST (deploy)
Apr 07 17:15:12 myapp-server crontab[8770]: (deploy) REPLACE (deploy)
Apr 07 17:15:12 myapp-server monit[8771]: Skipping 'allow localhost' -- host resolved to [::ffff:127.0.0.1] which is present in ACL already
Apr 07 17:15:12 myapp-server monit[931]: 'puma_rails' restart on user request
Apr 07 17:15:12 myapp-server monit[931]: Monit daemon with PID 931 awakened
Apr 07 17:15:12 myapp-server monit[931]: Awakened by User defined signal 1
Apr 07 17:15:12 myapp-server monit[931]: Reinitializing Monit -- control file '/etc/monit/monitrc'
Apr 07 17:15:12 myapp-server monit[8774]: Skipping 'allow localhost' -- host resolved to [::ffff:127.0.0.1] which is present in ACL already
Apr 07 17:15:12 myapp-server monit[931]: 'delayed_job_rails-1' restart on user request
Apr 07 17:15:12 myapp-server monit[931]: Monit daemon with PID 931 awakened
Apr 07 17:15:12 myapp-server monit[931]: Skipping 'allow localhost' -- host resolved to [::ffff:127.0.0.1] which is present in ACL already
Apr 07 17:15:12 myapp-server monit[931]: 'myapp-server' Monit reloaded
Apr 07 17:15:12 myapp-server systemd[1]: Reloading nginx - high performance web server.
```

### Custom JSON
```json
"appserver": {
    "dot_env": true,
    "worker_processes": 2
},
"worker": {
    "adapter": "delayed_job",
    "process_count": 1
}
```

### Chef log including error

**Comment by kzkn** [comment]
This seems to be similar to the problem we addressed in #251.
Enviroment variables are embed in monitrc files. So environment variables change causes `monit reload`. If appserver and worker is used in a instance both, `monit reload` is called twice. One for appserver's monitrc, one for worker's monitrc.

In the log, the second `Reinitializing Monit -- control file '/etc/monit/monitrc'` cancels `'puma_rails' restart on user request`.

**Comment by ajgon** [comment]
I created a PR #282 to address this issue, can you please take a look if it works for you? If so, I'll merge it to the codebase and release a new version of the gem.

Thanks!

**Comment by kzkn** [comment]
@ajgon #282 does not work :cry: And I found a new issue #284.
Even if `pumactl restart` is successfully to run, it will not resolve #284.

I think it is not necessary to embed environment variables in `monitrc`.

If we are using `dot_env`, it [does not override environment variables](https://github.com/bkeepers/dotenv/blob/c237d6d6291c898d8affb290b510c7aac49aed71/lib/dotenv/environment.rb#L21) already defined by default. If we set values in monitrc, puma worker processes will not reflect it even if the `.env` value changes.

If we do not include environment variables in monitrc, changes in `.env` values will be reflected to puma worker processes.

And puma's monitrc is not changed before/after the deploy which has the change of environment variables, it avoids `monit reload` for puma. Therefore, puma will restart even after a deployment in which environment variables have been changed. In other words, this issue (#281) is resolved.

My patch is here: https://github.com/kzkn/opsworks_ruby/commit/ba95731345916851c517f966f8dcac7af12d4aec
